In [1]:
import pandas as pd
import glob

In [2]:
path = r'C:\workspace\notebooks\health-ins-cross-sell\results'
all_files = glob.glob(path + "/*.csv")

In [3]:
summary_df = [pd.read_csv(f) for f in all_files]
summary_df_all=pd.concat(summary_df)
summary_df_all = summary_df_all.reset_index()
summary_df_all.drop('index',inplace=True,axis=1)

In [5]:
# total : Customer count treshold
# pct : Success percentage treshold
# col_count : Attribute complexity (1-4)
summary_df_all[(summary_df_all['total']>1000) & \
               (summary_df_all['pct']>.2)  & \
               (summary_df_all['col_count']==2) & \
               (summary_df_all['merged_columns_names'].apply(lambda x : 'Policy_Sales_Channel' not in x))  & \
               (summary_df_all['merged_columns_names'].apply(lambda x : 'Region_Code' not in x))]\
               .sort_values(by='pct',ascending=False)\
               .head(20)

merged_columns_names merged_columns_values        Y  \
3537175          Vehicle_Age >>> premium_group  > 2 Years >>> prem_4    514.0   
3537185              Vehicle_Age >>> age_group   > 2 Years >>> age_2   1339.0   
3537186              Vehicle_Age >>> age_group   > 2 Years >>> age_3   2254.0   
3538075           Vehicle_Damage >>> age_group         Yes >>> age_2  19813.0   
3537174          Vehicle_Age >>> premium_group  > 2 Years >>> prem_3   2271.0   
95767                   Gender >>> Vehicle_Age    Male >>> > 2 Years   3179.0   
3535955       Previously_Insured >>> age_group           0 >>> age_2  20105.0   
2430846     Previously_Insured >>> Vehicle_Age       0 >>> > 2 Years   4701.0   
606620         Driving_License >>> Vehicle_Age       1 >>> > 2 Years   4697.0   
3535970         Vehicle_Age >>> Vehicle_Damage     > 2 Years >>> Yes   4702.0   
3537173          Vehicle_Age >>> premium_group  > 2 Years >>> prem_2   1299.0   
3538076           Vehicle_Damage >>> age_group         Yes >>> age_3  13217.0   
95764                   Gender >>> Vehicle_Age  Female >>> > 2 Years   1523.0   
3535966         Vehicle_Age >>> Vehicle_Damage      1-2 Year >>> Yes  34210.0   
3538069       Vehicle_Damage >>> premium_group        Yes >>> prem_4   2347.0   
3535956       Previously_Insured >>> age_group           0 >>> age_3  13361.0   
2430844     Previously_Insured >>> Vehicle_Age        0 >>> 1-2 Year  34711.0   
3388597   Previously_Insured >>> premium_group          0 >>> prem_4   2363.0   
3538068       Vehicle_Damage >>> premium_group        Yes >>> prem_3  15700.0   
3074628  Previously_Insured >>> Vehicle_Damage             0 >>> Yes  45642.0   

                N     total       pct  col_count  
3537175     933.0    1447.0  0.355418          2  
3537185    2532.0    3871.0  0.345985          2  
3537186    4645.0    6899.0  0.326764          2  
3538075   42328.0   62141.0  0.318845          2  
3537174    4900.0    7171.0  0.316743          2  
95767      7121.0   10300.0  0.308678          2  
3535955   46092.0   66197.0  0.303721          2  
2430846   11260.0   15961.0  0.294556          2  
606620    11254.0   15951.0  0.294490          2  
3535970   11290.0   15992.0  0.294048          2  
3537173    3326.0    4625.0  0.280960          2  
3538076   36271.0   49488.0  0.267084          2  
95764      4184.0    5707.0  0.266947          2  
3535966   94015.0  128225.0  0.266800          2  
3538069    6532.0    8879.0  0.264385          2  
3535956   38248.0   51609.0  0.258898          2  
2430844  100280.0  134991.0  0.257139          2  
3388597    6931.0    9294.0  0.254303          2  
3538068   46844.0   62544.0  0.251031          2  
3074628  136849.0  182491.0  0.250108          2

## Notes


##### Quantitative Customer Personas for Health Insurance Cross Sell

* Average Cross sell rate is 12.4%

##### Indicators for increased likelyhood of cross sell
* Without previous insurance
* With Vehicle damage
* Between 32 and 46 years old
* Vehicle over 2 years old

##### Indicators for low likelyhood of cross sell
* No Vehicle damage
* Previously insured
* Vehicle below 1 year old
* 20 to 32 years old or 60 to 85 years old
* Female

##### Cross sell opportunities
* Customers between 32 and 46 years old with damaged vehicles (~31%)
* Customers between 32 and 46 years without previous insurance (~30%)
* Customers between 32 and 46 years without previous insurance with damaged vehicles (~33%)
* Customers between 32 and 46 with damaged vehicles paying premium between USD36292.0 and USD57361.0 (~33%)
* Customers between 32 and 60 with vehicles older than 2 years (~33%)
* Male customers with vehicles over 2 years old
* Customers with vechicles over 2 years old with premium above USD57361.0 (~35%)
* Customers between 32 to 46 years old with license but without previous insurance (~30%)
* Customers between 32 to 46 years old with damaged vehicles below 1 year old (~30%)